packages/commerce-sdk-react/src/hooks/ShopperBaskets/cache.ts

```diff
@@ -146,39 +72,287 @@ const TODO = (method: keyof Client) => {
 }
 
 export const cacheUpdateMatrix: CacheUpdateMatrix<Client> = {
-    addCouponToBasket: updateBasket,
+    addCouponToBasket(customerId, {parameters}, response) {
+        return {
+            update: [
+                {queryKey: getBasket.queryKey(parameters)},
+                ...(customerId
+                    ? [
+                          {
+                              queryKey: getCustomerBaskets.queryKey({...parameters, customerId}),
+                              updater: (oldData: CustomerBasketsResult | undefined) =>
+                                  customerBasketsUpdater(parameters, response, oldData)
+                          }
+                      ]
+                    : [])
+            ]
+        }
+    },
     addGiftCertificateItemToBasket: TODO('addGiftCertificateItemToBasket'),
-    addItemToBasket: updateBasket,
-    addPaymentInstrumentToBasket: updateBasket,
+    addItemToBasket(customerId, {parameters}, response) {
+        return {
+            update: [
+                {queryKey: getBasket.queryKey(parameters)},
+                ...(customerId
+                    ? [
+                          {
+                              queryKey: getCustomerBaskets.queryKey({...parameters, customerId}),
+                              updater: (oldData: CustomerBasketsResult | undefined) =>
+                                  customerBasketsUpdater(parameters, response, oldData)
+                          }
+                      ]
+                    : [])
+            ]
+        }
+    },
+    addPaymentInstrumentToBasket(customerId, {parameters}, response) {
+        return {
+            update: [
+                {queryKey: getBasket.queryKey(parameters)},
+                ...(customerId
+                    ? [
+                          {
+                              queryKey: getCustomerBaskets.queryKey({...parameters, customerId}),
+                              updater: (oldData: CustomerBasketsResult | undefined) =>
+                                  customerBasketsUpdater(parameters, response, oldData)
+                          }
+                      ]
+                    : [])
+            ]
+        }
+    },
     addPriceBooksToBasket: TODO('addPriceBooksToBasket'),
     addTaxesForBasket: TODO('addTaxesForBasket'),
     addTaxesForBasketItem: TODO('addTaxesForBasketItem'),
-    createBasket: updateBasketWithResponseBasketId,
+    createBasket(customerId, {parameters}, response) {
+        const {basketId} = response
+
+        return {
+            update: [
+                {queryKey: getBasket.queryKey(parameters)},
+                ...(customerId && basketId
+                    ? [
+                          {
+                              // @ts-ignore
+                              queryKey: getCustomerBaskets.queryKey({
+                                  ...parameters,
+                                  basketId,
```

Comment:
```suggestion
                              queryKey: getCustomerBaskets.queryKey({
                                  ...parameters,
```
The error is just that you're passing `basketId`, but you don't need to, because the endpoint doesn't use it.